### DISCUSSION TITLE
opening a component to mount with spotlight click

### DISCUSSION THREAD

**Discussion by bphoebew** [gilbarbara/react-joyride#928] [Q&A] [answered]
Is there any other ways to go about interacting with components without utilizing the controlled version? For instance, I have one step which is a button. This button opens a tooltip which contains the next step. I would rather not use props like in the example (modalOpen) because I have multiple tours configured to a single Joyride. 

I tried utilizing spotlightClicks which opens the tooltip. However, the tour cannot recognize that the tooltip is now mounted so it ignores that step and goes onto the next one.

**Comment by gilbarbara** [comment]
Hey @bphoebew 
As far as I'm aware the only way this can be achieved would be to control the stepIndex to advance to the next step when it is available.

**Reply by bphoebew** [reply]
Can you elaborate on what you mean by this?

**Reply by bphoebew** [reply]
The spotlight click allows me to click on it (meaning that it is now available). But it still goes onto the next step

**Reply by gilbarbara** [reply, marked as answer]
It means I think the only way to control the steps manually is with the controlled mode.

To make sure I understand correctly:
You have a step where the target is a button, and you enabled the spotlightClick for it so that you can click on the button.
The button opens a tooltip with the next step, but since the tooltip with the target isn't rendered when you click, it got skipped.
Is that it?

**Reply by bphoebew** [reply]
Yes, the tooltip is not rendered at first. However, the tooltip does popup with the button (spotlight) click. So, are you saying that because it was not part of the initial rendering, it does not work?
